source/Xunit.Gherkin.Quick.UnitTests/GherkinScenarioDefinitionTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+using Xunit.Gherkin.Quick;
+
+namespace UnitTests
+{
+    public class GherkinScenarioDefinitionTests
+    {
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void ApplyBackground_WithNoBackground_CopiesScenarioDetails(Gherkin.Ast.ScenarioDefinition baseDefinition)
+        {
+            Gherkin.Ast.ScenarioDefinition modified = null;
+
+            if (baseDefinition is Gherkin.Ast.Scenario scenario)
+                modified = scenario.ApplyBackground(null);
+            else if (baseDefinition is Gherkin.Ast.ScenarioOutline scenarioOutline)
+                modified = scenarioOutline.ApplyBackground(null);
```

Comment:
Okay, I see you defined the extension method on the outline as well. I think this is not needed. My point was to only implement it for the scenario. ScenarioOutlineExecutor can call the method on the scenario, which it acquires from `ScenarioOutline.ApplyExampleRow()` call (which converts outline into scenario).
So, I think you can remove this method from outline. Scenario having it is enough and reusable for both scenario and outline execution.